tensorboard/plugins/hparams/_keras.py

```diff
@@ -62,10 +62,12 @@ def __init__(self, writer, hparams, trial_id=None):
         summary_v2.hparams_pb(self._hparams, trial_id=self._trial_id)
         if writer is None:
             raise TypeError(
-                "writer must be a `SummaryWriter` or `str`, not None"
+                "writer must be a `SummaryWriter`, `str` or `PathLike`, not None"
             )
         elif isinstance(writer, str):
             self._writer = tf.compat.v2.summary.create_file_writer(writer)
+        elif isinstance(writer, os.PathLike):
+            self._writer = tf.compat.v2.summary.create_file_writer(os.fsdecode(writer))
```

Comment:
`os.fsdecode` seems to be related/useful for file names with specific encodings, but we'd prefer not to get into anything related to filesystem encodings for now. Can you please update it to use `os.fspath`?

TensorFlow does not use `os.fsdecode` anywhere, but it uses `os.fspath` in several places, so that would also be more consistent.

Reply:
Good point, sorry for the additional work :)